onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2486,208 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // validating attributes shape
+  // Original image has shape either [N, C, H, W] or [C, H, W]
+  //  but N and C are ignored for pads, dilations and stride computation
+  size_t n_input_dims = 2;
+
+  std::vector<int64_t> pads;
+  if (getRepeatedAttribute(ctx, "pads", pads)) {
+    if ((pads.size() != 0) && (pads.size() != n_input_dims * 2)) {
+      fail_shape_inference("Attribute pads has incorrect size");
+    }
+  }
+
+  std::vector<int64_t> dilations;
+  if (getRepeatedAttribute(ctx, "dilations", dilations)) {
+    if ((dilations.size() != 0) && (dilations.size() != n_input_dims)) {
+      fail_shape_inference("Attribute dilations has incorrect size");
+    }
+  }
+
+  std::vector<int64_t> strides;
+  if (getRepeatedAttribute(ctx, "strides", strides)) {
+    if ((strides.size() != 0) && (strides.size() != n_input_dims)) {
+      fail_shape_inference("Attribute strides has incorrect size");
+    }
+  }
+
+  // Validating inputs - all inputs are required
+  if (!hasNInputShapes(ctx, 3)) {
+    fail_shape_inference("Missing input");
```

Comment:
This should be just `return`. Input shapes may not be statically known, and that is not a failure condition.